### PR TITLE
Transaction list suggestion

### PR DESCRIPTION
**Pull request by JoranHonig**
This pr proposes some changes to the feature branch for transaction sequence outputting.

It separates different logics into different functions and adds documentation

### CODE REVIEW

File: mythril/analysis/report.py [hunk 1]
```diff
@@ -75,14 +76,16 @@ def transaction_sequence_users(self):
 
     @property
     def transaction_sequence_jsonv2(self):
+        """ Returns the transaction sequence in json with pre-generated block data"""
         return (
-            self.add_block_data(self.transaction_sequence)
+            json.dumps(self.add_block_data(self.transaction_sequence), ident=4)
```
Comment (norhh) [inline mythril/analysis/report.py]: is it ident or indent?

Comment (JoranHonig) [reply mythril/analysis/report.py]: indent 😅

**Comment by norhh** [comment]
Hey, This can make the PR quite big(as this changes a big part of solver.py, as the PR has become quite big due to the resolution of various merge conflicts, this suggestion can be added separately,  what do you think?

**Comment by JoranHonig** [comment]
The main reason why I created this pr is bc it makes it easier to review the added business logic.
Additionally, there are some changes added in this pr that fix some small errors

**Pull request merged**